### PR TITLE
Web tier lombok

### PR DESCRIPTION
**Pull request by allga**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -98,6 +98,13 @@
                 <artifactId>spring-context</artifactId>
                 <version>${spring.version}</version>
             </dependency>
+            <dependency>
```
Comment (athlon0x01) [inline pom.xml]: Do we need lombok in root pom file?

File: web-tier/pom.xml [hunk 2]
```diff
@@ -74,6 +74,11 @@
             <version>${spring-boot.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
```
Comment (athlon0x01) [inline web-tier/pom.xml:79]: please add version

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
 import javax.validation.constraints.Min;
 import javax.validation.constraints.PositiveOrZero;
 import java.util.Objects;
 
-public class DishProduct {
+@AllArgsConstructor
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java]: https://projectlombok.org/features/Data

AllArgsConstructor in included in Data, please try to remove it
and double check if NoArgsConstructor requires explicit declaration

and add Builder

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java [hunk 4]
```diff
@@ -1,25 +1,18 @@
...
+@NoArgsConstructor
+public @Data class DishProduct {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java]: I prefer to put class level annotation above the class definition line
```
@Data
public class DishProduct {

File: web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java [hunk 5]
```diff
@@ -47,12 +47,12 @@ public List<CategoryWithDishes> getAllDishes() {
         return categories.stream()
                 .map(c -> {
                     final CategoryWithDishes cView = new CategoryWithDishes();
-                    cView.id = c.id;
-                    cView.name = c.name;
-                    final List<Dish> dishList = dishesMap.get(c.id);
-                    cView.dishes = (dishList == null) ? new ArrayList<>() : dishList.stream()
+                    cView.setId(c.getId());
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java]: use Builder for building complex objects

File: web-tier/src/main/java/com/outdoor/foodcalc/model/product/ProductView.java [hunk 6]
```diff
@@ -1,57 +1,28 @@
...
 /**
  * Simplified view model for {@link com.outdoor.foodcalc.domain.model.product.Product} class.
  *
  * @author Anton Borovyk
  */
-public class ProductView {
-    public long id;
+@AllArgsConstructor
+@NoArgsConstructor
+public @Data class ProductView {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/product/ProductView.java]: add Builder

File: web-tier/src/main/java/com/outdoor/foodcalc/model/product/CategoryWithProducts.java [hunk 7]
```diff
@@ -1,14 +1,20 @@
...
  *
  * @author Anton Borovyk
  */
-public class CategoryWithProducts {
-    public long id;
-    public String name;
-    public List<ProductView> products;
+@AllArgsConstructor
+@NoArgsConstructor
+public @Data class CategoryWithProducts {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/product/CategoryWithProducts.java]: add Builder

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java [hunk 8]
```diff
@@ -1,28 +1,21 @@
...
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotEmpty;
 import java.util.List;
 import java.util.Objects;
 
-public class SimpleDish {
-    public long id;
+@AllArgsConstructor
+@NoArgsConstructor
+public @Data class SimpleDish {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java]: add Builder

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishView.java [hunk 9]
```diff
@@ -1,23 +1,27 @@
...
 
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotEmpty;
 import java.util.List;
-
-public class DishView {
-    public long id;
+ @AllArgsConstructor
+ @NoArgsConstructor
+public @Data class DishView {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishView.java]: Add Builder

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java [hunk 10]
```diff
@@ -1,9 +1,15 @@
...
+
 import java.util.List;
 
-public class CategoryWithDishes {
-    public long id;
-    public String name;
-    public List<DishView> dishes;
+@AllArgsConstructor
+@NoArgsConstructor
+public @Data class CategoryWithDishes {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java]: Add Builder

File: web-tier/src/test/java/com/outdoor/foodcalc/service/DishCategoryServiceTest.java [hunk 11]
```diff
@@ -23,53 +24,52 @@
  */
 public class DishCategoryServiceTest {
 
+    @Rule
```
Comment (athlon0x01) [inline web-tier/src/test/java/com/outdoor/foodcalc/service/DishCategoryServiceTest.java]: Leave is as it was.
You'll have separate task to use Junit5

**Review by athlon0x01 — CHANGES_REQUESTED** [review]

File: web-tier/pom.xml (hunk 2 above)
Comment (allga) [reply web-tier/pom.xml:79]: done

File: pom.xml (hunk 1 above)
Comment (allga) [reply pom.xml]: lombok deleted from root pom file

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java (hunk 3 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java]: they both require an explicit declaration, there are errors without any of them

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java (hunk 4 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java]: done

File: web-tier/src/test/java/com/outdoor/foodcalc/service/DishCategoryServiceTest.java (hunk 11 above)
Comment (allga) [reply web-tier/src/test/java/com/outdoor/foodcalc/service/DishCategoryServiceTest.java]: done

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java (hunk 3 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishProduct.java]: Builder added

File: web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java (hunk 5 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java]: Builder added

File: web-tier/src/main/java/com/outdoor/foodcalc/model/product/CategoryWithProducts.java (hunk 7 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/product/CategoryWithProducts.java]: Builder added

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java (hunk 8 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java]: Builder added

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishView.java (hunk 9 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/DishView.java]: Builder added

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java (hunk 10 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java]: Builder added

File: pom.xml [hunk 12]
```diff
@@ -98,6 +98,7 @@
                 <artifactId>spring-context</artifactId>
                 <version>${spring.version}</version>
             </dependency>
+
```
Comment (athlon0x01) [inline pom.xml]: please remove extra line.
Try revert to make file the same as before commit

File: web-tier/pom.xml (hunk 2 above)
Comment (athlon0x01) [reply web-tier/pom.xml:79]: please move version to root pom, like spring-boot.version

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java [hunk 13]
```diff
@@ -1,9 +1,15 @@
 package com.outdoor.foodcalc.model.dish;
 
-import java.util.List;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.jackson.Jacksonized;
 
+import java.util.List;
+@Data
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java:9]: please add one line between import and annotations

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java [hunk 14]
```diff
@@ -1,28 +1,20 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.jackson.Jacksonized;
+
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotEmpty;
 import java.util.List;
-import java.util.Objects;
-
+@Data
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java:11]: please add one line between import and annotations

File: web-tier/src/main/java/com/outdoor/foodcalc/service/DishCategoryService.java [hunk 15]
```diff
@@ -43,9 +43,7 @@ public List<SimpleDishCategory> getDishCategories() {
     }
 
     private SimpleDishCategory mapDishCategory(DishCategory category) {
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/DishCategoryService.java:45]: remove local variable, just return created object

File: web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java [hunk 16]
```diff
@@ -46,14 +46,14 @@ public List<CategoryWithDishes> getAllDishes() {
...
                 .map(c -> {
-                    final CategoryWithDishes cView = new CategoryWithDishes();
-                    cView.id = c.id;
-                    cView.name = c.name;
-                    final List<Dish> dishList = dishesMap.get(c.id);
-                    cView.dishes = (dishList == null) ? new ArrayList<>() : dishList.stream()
-                            .map(this::mapDishView)
-                            .collect(Collectors.toList());
-                    return cView;
+                    final List<Dish> dishList = dishesMap.get(c.getId());
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/DishService.java:53]: Lets move all `SimpleDishCategory` to separate function

File: web-tier/src/main/java/com/outdoor/foodcalc/service/MealTypesService.java [hunk 17]
```diff
@@ -43,9 +43,7 @@ public List<MealTypeView> getMealTypes() {
     }
 
     private MealTypeView mapMealType(MealType mealType) {
-        final MealTypeView model = new MealTypeView();
-        model.id = mealType.getTypeId();
-        model.name = mealType.getName();
+        final MealTypeView model = new MealTypeView(mealType.getTypeId(), mealType.getName());
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/MealTypesService.java]: remove local variable, just return created object

File: web-tier/src/main/java/com/outdoor/foodcalc/service/ProductCategoryService.java [hunk 18]
```diff
@@ -58,9 +58,8 @@ public SimpleProductCategory getCategory(long id) {
     }
 
     private SimpleProductCategory mapProductCategory(ProductCategory productCategory) {
-        final SimpleProductCategory model = new SimpleProductCategory();
-        model.id = productCategory.getCategoryId();
-        model.name = productCategory.getName();
+        final SimpleProductCategory model = new SimpleProductCategory(
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/ProductCategoryService.java]: remove local variable, just return created object

File: web-tier/src/main/java/com/outdoor/foodcalc/service/ProductService.java [hunk 19]
```diff
@@ -61,14 +61,13 @@ public List<CategoryWithProducts> getAllProducts() {
...
-        withProducts.id = category.getCategoryId();
-        withProducts.name = category.getName();
         final List<Product> productList = productsMap.get(category.getCategoryId());
-        withProducts.products = (productList == null) ? new ArrayList<>() : productList.stream()
-                .map(this::mapSimpleProduct)
-                .collect(Collectors.toList());
-        return withProducts;
+        return CategoryWithProducts.builder()
+                .id(category.getCategoryId())
+                .name(category.getName()).products((productList == null) ? new ArrayList<>() : productList.stream()
```
Comment (athlon0x01) [inline web-tier/src/main/java/com/outdoor/foodcalc/service/ProductService.java]: please move product to new line

**Review by athlon0x01 — CHANGES_REQUESTED** [review]

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java (hunk 13 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/CategoryWithDishes.java:9]: done

File: web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java (hunk 14 above)
Comment (allga) [reply web-tier/src/main/java/com/outdoor/foodcalc/model/dish/SimpleDish.java:11]: done

**Review by athlon0x01 — APPROVED** [review]

**Pull request merged**